pkg/operator/resource_reconciler.go

```diff
@@ -336,7 +338,7 @@ func (rr *ResourceReconciler) OnUpdate(old, cur interface{}) {
 		return
 	}
 
-	if rr.DeletionInProgress(mCur) {
+	if !k8sutil.IsFinalizerPresent(mCur.GetFinalizers(), k8sutil.StatusCleanupFinalizerName) && rr.DeletionInProgress(mCur) {
```

Comment:
maybe we should have a new method which returns true if the finalizer is not set and the deletion timestamp is not nil?